File: Project5/src/read_data.R

#read data
step = 24
datafile <- paste0('../data/Project5_fobs_jja2022_step',step,'_00.csv')
climfile <- paste0('../data/Project5_clim_jja2022_step',step,'_00.csv')

data <- read.csv(datafile, header = TRUE)
clim <- read.csv(climfile, header = TRUE)


#compute SEEPS
seeps_mtx = clim[6:14]

ob_ind <- (data["OBS"] > clim["t1"])*1 + (data["OBS"]  >= clim["t2"])*1
fc_ind <- (data["FCT_ifs"] > clim["t1"])*1 + (data["FCT_ifs"]  >= clim["t2"])*1

indices <- fc_ind * 3 + ob_ind + 1

seeps_all <- array(dim=c(length(indices)))
for ( jj in 1:length(indices) ){
    if (! is.na(indices[jj]) ) {
        seeps_all[jj] <- seeps_mtx[jj,indices[jj]]
    }
}
mean(seeps_all,na.rm=TRUE)

File: Project5/src/read_data.py
import glob
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

datapath = '../data/Project5_fobs_jja2022_step{step}_00.csv'
climpath = '../data/Project5_clim_jja2022_step{step}_00.csv'

step = 24

df_fobs = pd.read_csv(datapath.format(step=step))
df_clim = pd.read_csv(climpath.format(step=step))

#SEEPS
# local 3x3 SEEPS matrices
seeps_mtx = np.zeros((len(df_clim),9))
for ii in range(9):
    seeps_mtx[:,ii] = df_clim[f"value_{ii}"]

#SEEPS for FCT_ifs
ob_ind = (df_fobs["OBS"] > df_clim["t1"]).astype(int) + (df_fobs["OBS"]  >= df_clim["t2"]).astype(int)
fc_ind = (df_fobs["FCT_ifs"] > df_clim["t1"]).astype(int) + (df_fobs["FCT_ifs"]  >= df_clim["t2"]).astype(int)

indices = fc_ind * 3 + ob_ind
seeps_all = np.array([ seeps_mtx[jj,idx] for jj,idx in enumerate(indices)])
seeps = np.nanmean(seeps_all,axis=0)
